src/main/java/unibook/model/module/group/Group.java

```diff
@@ -0,0 +1,134 @@
+package unibook.model.module.group;
+
+import static java.util.Objects.requireNonNull;
+
+import java.time.LocalDateTime;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import unibook.model.module.Module;
+import unibook.model.module.exceptions.DuplicateMeetingTimeException;
+import unibook.model.module.exceptions.MeetingTimeNotFoundException;
+import unibook.model.person.Student;
+import unibook.model.person.exceptions.PersonNotFoundException;
+
+/**
+ * Represents a group of students within a module.
+ */
+public class Group {
+    private final String name;
+    private final Module module;
+    private final ObservableList<Student> members;
+    private final ObservableList<LocalDateTime> meetingTimes;
+
+    /**
+     * Instantiates a group object.
+     *
+     * @param name         of the group.
+     * @param module       that the group is in.
+     * @param members      of the group.
+     * @param meetingTimes meeting times of the group.
+     */
+    public Group(String name, Module module, ObservableList<Student> members,
+                 ObservableList<LocalDateTime> meetingTimes) {
```

Comment:
Can consider checking `requireAllNonNull` here to ensure no null fields are passed to constructor